src/features/filter.ts

```diff
@@ -1,12 +1,29 @@
-import { createSlice } from '@reduxjs/toolkit';
+import { createSlice, PayloadAction, Slice } from '@reduxjs/toolkit';
+import { Status } from '../types/Status';
+
+type FilterState = {
+  query: string;
+  status: Status;
+};
 
 const initialState = {
   query: '',
-  status: 'all',
+  status: Status.All,
 };
 
-export const filterSlice = createSlice({
+export const filterSlice: Slice<FilterState> = createSlice({
   name: 'filter',
   initialState,
-  reducers: {},
+  reducers: {
+    setStatus: (state, { payload }: PayloadAction<Status>) => ({
+      ...state,
+      status: payload,
+    }),
+    setQuery: (state, { payload }: PayloadAction<string>) => ({
+      ...state,
+      query: payload,
```

Comment:
The reducer functions `setStatus` and `setQuery` are returning new state objects. Since Redux Toolkit uses Immer under the hood, you can directly mutate the state object. For example, you can simply write `state.status = payload` and `state.query = payload` without returning a new object.